pkg/components/velero/csi/drivers/aws/aws.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package aws deals with configuring Velero aws plugin.
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/kinvolk/lokomotive/internal"
+)
+
+const indentation = 6
+
+// Configuration contains AWS specific parameters.
+type Configuration struct {
+	Credentials            string                  `hcl:"credentials"`
+	BackupStorageLocation  *BackupStorageLocation  `hcl:"backup_storage_location,block"`
+	VolumeSnapshotLocation *VolumeSnapshotLocation `hcl:"volume_snapshot_location,block"`
+}
+
+// BackupStorageLocation configures the backup storage location for AWS plugin.
+type BackupStorageLocation struct {
+	Region           string `hcl:"region"`
+	Bucket           string `hcl:"bucket"`
+	Name             string `hcl:"name,optional"`
+	Prefix           string `hcl:"prefix,optional"`
+	S3ForcePathStyle bool   `hcl:"s3_force_path_style,optional"`
+	S3URL            string `hcl:"s3_url,optional"`
+	PublicURL        string `hcl:"public_url,optional"`
+}
+
+// validate validates BackupStorageLocation struct fields.
+func (b *BackupStorageLocation) validate() hcl.Diagnostics { // nolint:dupl
```

Comment:
what do you mean ?

Reply:
I mean:
```
$ git grep 'BackupStorageLocation) validate'
pkg/components/velero/csi/drivers/aws/aws.go:func (b *BackupStorageLocation) validate() hcl.Diagnostics {
pkg/components/velero/openebs/openebs.go:func (b *BackupStorageLocation) validate(defaultProvider string) hcl.Diagnostics { // nolint:dupl
```

This code is very similar, we can perhaps put it in separate package and import to avoid duplication.